daemon/anonymous-statistics.sh

```diff
@@ -0,0 +1,141 @@
+# - KICKSTART64 - installed using the kickstart-static64.sh script
+# - MAKESELF    - installed by running a makeself package
+# - INSTALL     - installed using netdata-installer.sh
+# - UPDATE      - updated using netdata-updater.sh
+# - FATAL       - netdata exited due to a fatal condition
+# - START       - netdata started
+
+ACTION="${1}"
+ACTION_RESULT="${2}"
+ACTION_DATA="${3}"
+
+# -------------------------------------------------------------------------------------------------
+# check opt-out
+
+OPTOUT=
+if [ -z "${NETDATA_CONFIG_DIR}" ]; then
+    for dir in "$(netdata -W get global 'config directory' '/etc/netdata' 2>/dev/null)" "/etc/netdata" "/opt/netdata/etc/netdata"; do
+        if [ -f "${dir}/.opt-out-from-anonymous-statistics" ]; then
+            OPTOUT="$(cat "${dir}/.opt-out-from-anonymous-statistics")"
+            break
+        fi
+    done
+else
+    OPTOUT="$(cat "${NETDATA_CONFIG_DIR}/.opt-out-from-anonymous-statistics")"
+fi
+
+if [ "${OPTOUT}" = "all" ] || ( [ "${OPTOUT}" = "usage" ] && [ "${ACTION}" = "START" ] ); then
+    # the user does not want us to send statistics
+    exit 0
+fi
+
+# -------------------------------------------------------------------------------------------------
+# detect the operating system
+
+OS_DETECTION=
+NAME=
+VERSION=
+VERSION_ID=
+ID=
+ID_LIKE=
+
+if [ -f "/etc/os-release" ]; then
+    OS_DETECTION="/etc/os-release"
+    eval "$(grep -E "^(NAME|ID|ID_LIKE|VERSION|VERSION_ID)=" </etc/os-release)"
+elif [ -f "/etc/lsb-release" ]; then
+    OS_DETECTION="/etc/lsb-release"
+    DISTRIB_ID=
+    DISTRIB_RELEASE=
+    DISTRIB_CODENAME=
+    eval "$(grep -E "^(DISTRIB_ID|DISTRIB_RELEASE|DISTRIB_CODENAME)=" </etc/lsb-release)"
+    NAME="${DISTRIB_ID}"
+    VERSION="${DISTRIB_RELEASE}"
+    ID="${DISTRIB_CODENAME}"
+elif [ ! -z "$(command -v lsb_release 2>/dev/null)" ]; then
+    OS_DETECTION="lsb_release"
+    NAME="$(lsb_release -i 2>/dev/null | sed "s/[[:space:]]*:[[:space:]]*/:/g" | cut -d ':' -f 2)"
+    VERSION_ID="$(lsb_release -f 2>/dev/null | sed "s/[[:space:]]*:[[:space:]]*/:/g" | cut -d ':' -f 2)"
+    VERSION="$(lsb_release -c 2>/dev/null | sed "s/[[:space:]]*:[[:space:]]*/:/g" | cut -d ':' -f 2)"
+fi
+
+# -------------------------------------------------------------------------------------------------
+# detect the kernel
+
+KERNEL_NAME="$(uname -s)"
+KERNEL_VERSION="$(uname -r)"
+ARCHITECTURE="$(uname -m)"
+
+# -------------------------------------------------------------------------------------------------
+# detect the virtualization
+
+VIRTUALIZATION="unknown"
+VIRT_DETECTION="none"
+
+if [ ! -z "$(command -v systemd-detect-virt 2>/dev/null)" ]
+then
+    VIRTUALIZATION="$(systemd-detect-virt)"
+    VIRT_DETECTION="systemd-detect-virt"
+else
+    # /proc/1/sched exposes the host's pid of our init !
+    # http://stackoverflow.com/a/37016302
+    pid=$( head -n 1 </proc/1/sched 2>/dev/null | { IFS='(),#:' read name pid th threads; echo $pid; } )
```

Comment:
unused variables: `name`, `th`, `threads`
use `read -r` instead of `read` to prevent mangling backslashes